toolkit/features/k8s_kubeconfig.py

```
@@ -107,18 +112,58 @@ def output_path(env: str) -> Path:
+    failure always raises immediately — the tunnel won't fix a key problem.
+
+    Interactive key auth (passphrase / ssh-agent) is still required; this runs
+    from an operator shell, not an unattended one.
     """
     access = load_cluster_access(env)
     logger.section(f"Fetch kubeconfig — {env} (node {access.node}, ssh {access.ssh_alias})")
 
-    result = command.run_list(fetch_argv(access.ssh_alias))
-    rewritten = rewrite_server(result.stdout, access.local_port)
+    argv = fetch_argv(access.ssh_alias)
+    result = subprocess.run(argv, capture_output=True, text=True)
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)